### DISCUSSION TITLE
Unable to parse certain url's as link in react-markdown

### DISCUSSION THREAD

**Discussion by nikhil19kekan** [remarkjs/remark#578] [Q&A] [answered]
using react markdown with gfm plugin, gfm plugin is unable to identify it as link

```
import ReactMarkdown from 'react-markdown';
const text = "http://azure/webapp/index?view=tev&cio=TEV";

<ReactMarkdown plugins={[gfm]} linkTarget={"_blank"} source={text} allowDangerousHtml />
```

even tried 

```
import ReactMarkdown from 'react-markdown';
const text = "http://azure/webapp/index?view=tev&cio=TEV";

<ReactMarkdown renderer ={{link:props => <a href={props.href} target="_blank" >{props.children}</a>}} linkTarget={"_blank"} source={text} allowDangerousHtml />
```

still the URL gets rendered as text and not as link

**Comment by ChristianMurphy** [comment, marked as answer]
@nikhil19kekan this is the expected behavior, from the GFM spec https://github.github.com/gfm/#valid-domain

> There must be at least one period

the example link you provided

```markdown
http://azure/webapp/index?view=tev&cio=TEV
```

does not include any periods, so autolinking will not see it as a valid link.

To make it a link, consider using
```markdown
<http://azure/webapp/index?view=tev&cio=TEV>
```

**Reply by nikhil19kekan** [reply]
thanks

**Comment by nikhil19kekan** [comment]
we want to maintain consistency among url's. so will it work if all url's are enclosed in "<" and ">" tag?

**Reply by wooorm** [reply]
`http://azure/webapp/index?view=tev&cio=TEV` is not a URL. It’s not valid.

There are many ways for creating links in markdown. [See the spec](https://spec.commonmark.org/0.29/). If you want almost all random values to “link”, use full inline links: `[text](<the value>)`

**Reply by nikhil19kekan** [reply]
thanks
